tasks/main.yml

```
@@ -28,8 +37,54 @@
     user_install: false
   with_items: "{{ fluentd_plugins }}"
 
+- name: retrieve initial ExecStart
+  shell: 'grep -i "ExecStart=.*/.*" /lib/systemd/system/multi-user.target.wants/td-agent.service | sed -r "s/^([^\/])*(\/.*)/\2/"'
+  register: initial_cmd
+  tags:
+  - never
+  - aggregator
+
+- name: Ensure unit directory is present
+  file:
+    path: /etc/systemd/system/td-agent.service.d
+    state: directory
+    mode: '0755'
+  tags:
+  - never
+  - aggregator
+
+- name: Ensure override file is absent
+  file:
+    path: /etc/systemd/system/td-agent.service.d/override.conf
+    state: absent
+  tags:
+  - never
+  - aggregator
+
+- name: reset ExecStart to null
+  blockinfile:
+    path: /etc/systemd/system/td-agent.service.d/override.conf
+    create: yes
+    block: |
+      [Service]
+      ExecStart=
+  tags:
+  - never
+  - aggregator
+
+- name: Add vault secret to ExecStart
+  lineinfile:
+    path: /etc/systemd/system/td-agent.service.d/override.conf
+    state: present
+    insertafter: '^ExecStart=$'
+    line: 'ExecStart=vault --config-file {{ fluentd_vault_config_path }} env -p {{ fluentd_secret_path }}{{ fluent_secret_name }} -- {{ initial_cmd.stdout }}'
```

To improve readability:
```suggestion
    line: > 
    'ExecStart=vault --config-file {{ fluentd_vault_config_path }}
    env -p {{ fluentd_secret_path }}{{ fluent_secret_name }} -- {{ initial_cmd.stdout }}'
```